boot/zephyr/CMakeLists.txt

```diff
@@ -252,6 +252,26 @@ if(CONFIG_MCUBOOT_SERIAL)
     )
 endif()
 
+if(CONFIG_SOC_ESP32 OR CONFIG_SOC_ESP32S2 OR CONFIG_SOC_ESP32C3 OR CONFIG_SOC_ESP32S3)
+
+  if(CONFIG_SOC_ESP32)
+    zephyr_compile_definitions(CONFIG_ESP32_XTAL_FREQ=40)
+  endif()
+  zephyr_compile_definitions(BOOTLOADER_BUILD=1)
+  zephyr_compile_definitions(CONFIG_MCUBOOT=1)
+  zephyr_compile_definitions(NDEBUG=1)
```

Comment:
If `BOOTLOADER_BUILD` and `NDEBUG` are fixed, could it be placed somewhere else? sdkconfig.h?